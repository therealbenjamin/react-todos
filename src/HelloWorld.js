import React, { Component } from 'react';
import './HelloWorld.css'

class HelloWorld extends Component {
  constructor(props) {
    super(props);
    this.state = { greeting: 'Hello' };
  }

  render() {
    return(
      <div className="HelloWorld">{this.state.greeting} {this.props.name}!</div>
    );
  }
}

export default HelloWorld;
