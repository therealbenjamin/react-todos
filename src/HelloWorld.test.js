import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorld from './HelloWorld';

describe(HelloWorld, () => {
  const name = 'Person';
  const mockRemoveGreeting = jest.fn();
  const component = shallow(
    <HelloWorld name={name} removeGreeting={mockRemoveGreeting}/ >
  )

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <HelloWorld name="Person"/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the name we supply', () => {
    debugger;
    expect(component.text()).toContain(name);
  });



});
