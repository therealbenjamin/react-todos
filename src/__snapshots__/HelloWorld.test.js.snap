// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorld extends _react.Component {
  constructor(props) {
    super(props);
    this.state = { greeting: 'Hello' };
    this.frenchify = this.frenchify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }

  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }

  render() {
    return (
      _react2.default.createElement('div', { className: 'HelloWorld', __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this },
        _react2.default.createElement('p', { __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this }, this.state.greeting, ' ', this.props.name, '!'),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this }),
        _react2.default.createElement('button', { onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }, 'Frenchify!'),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this }),
        _react2.default.createElement('button', { onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this }, 'Remove Me')));


  }

  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorld"
>
  <p>
    Hello
     
    Person
    !
  </p>
  <br />
  <button
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    Remove Me
  </button>
</div>
`;
